lib/elasticsearch_client.py

```diff
@@ -56,11 +56,21 @@ def create_index(self, index_name, ignore_fail=False, mapping=None):
             }'''
         self.es_connection.indices.create(index=index_name, update_all_types='true', body=mapping)
 
-    def create_alias(self, alias_name, index_name):
-        if self.es_connection.indices.exists_alias(index='*', name=alias_name):
-            self.es_connection.indices.delete_alias(index='*', name=alias_name)
-
-        self.es_connection.indices.put_alias(index=index_name, name=alias_name)
+#    def create_alias(self, alias_name, index_name):
+#        if self.es_connection.indices.exists_alias(index='*', name=alias_name):
+#            self.es_connection.indices.delete_alias(index='*', name=alias_name)
+#
+#        self.es_connection.indices.put_alias(index=index_name, name=alias_name)
+
+    def update_alias(self, oldindex, index, alias):
+        actions = None
+        if self.es_connection.indices.exists_alias(oldindex, alias):
+        # Add and remove aliases.
+            actions = [
+                {'remove': {'index': oldindex, 'alias': alias}},
```

Comment:
Might consider always having the remove match * for oldindex, since all we want is an atomic set the alias to be X. 

i.e. we don't really care what it is now, but we know what we want it to be, so we don't have to worry about what it currently may be.

Reply:
That makes sense, I always try to be as explicit as possible, but am ok with not being explicit where it's not needed.

Reply:
Done!